### ISSUE TITLE
Is TimeDistributed redundant?

### ISSUE THREAD

**Issue by tactycHQ** [keras-team/keras#11595]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by tinkermachine** [comment]
I have attempted to rework the Climate prediction example by Chollet in his book, Deep Learning with Python. Instead of simply predicting 1 variable 24 hours in the future (i.e. delay of 144 timesteps), I want to predict a sequence of 720 timesteps, 144 timesteps  in the future. That is I want 5 days hours worth of predictions, 24 hours from now.

My many-to-many model is as follows:

- 1 hidden GRU layer of 16 units with return_sequences=True. Input to LSTM is (None, 720, 14). i.e. timesteps is 720 and dimensions is 14. **return_sequences=true** for GRU layer so output of this layer is (None,720,16) as expected.

- Output layer is simply a Dense layer of 1 neuron which outputs a shape (None, 720, 1)

I have notices regardless of whether I use **regular Dense (1)** layer or a **TimeDistributed(Dense(1))** layer as the output, the loss and predictions are exactly the same. Also, the output shape from the Dense layer is (None, 720, 1) regardless of whether I use TimeDistributed(Dense(1)) or regular Dense(1).

My questions is if this is expected and if so, in which scenario should I expect to use the  TimeDistributed functionality?

My code can be found here: https://github.com/tinkermachine/chollet/blob/master/Climate/Climate%202.py
